src/lib/support/tests/TestPersistedCounter.cpp

```diff
@@ -35,12 +38,160 @@
 #include <nlunit-test.h>
 
 #include <CHIPVersion.h>
+#include <lib/support/Base64.h>
+#include <lib/support/CHIPArgParser.hpp>
 #include <lib/support/CHIPMem.h>
+#include <lib/support/CodeUtils.h>
 #include <lib/support/PersistedCounter.h>
 #include <lib/support/UnitTestRegistration.h>
+#include <platform/ConfigurationManager.h>
 #include <platform/PersistedStorage.h>
 
-#include "TestPersistedStorageImplementation.h"
+namespace
+{
+
+constexpr size_t kMaxPersistedValueLengthSupported = 256;
+
+std::map<std::string, std::string> sPersistentStore;
+
+FILE *sPersistentStoreFile = nullptr;
+
+} // namespace
+
```

Comment:
This code added is a just a combine of  TestPersistedStorageImplementation.h/cpp that was ONLY used by this test and is not applicable with newer usage.